csharp/ql/test/utils/model-generator/theorems/TheoremSummaries.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Linq;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Summaries;
+
+public class Theorems1<T> {
+    
+    public T Prop {
+        get { throw null; }
+        set { throw null; }
+    }
+
+    public Theorems1(T t)  { throw null; } 
+
+    public T Get() { throw null; }
+
+    public T Get(object x) { throw null; }
+
+    public T Id(T x) { throw null; }
+
+    public T Transform(int x, T y) { throw null; }
+
+    public S Transform<S>(S x) { throw null; }
+
+    public void Set(T x) { throw null; }
+
+    public void Set(int x, T y) { throw null; }
+
+    // No summary as S is unrelated to T
+    public void Set<S>(S x) { throw null; }
+
+    public int Apply(Func<T, int> f) { throw null; }
+}
```

Comment:
Also, I'd like the test case
```
Theorems1<S> FlatMap<S>(Func<T, IEnumerable<S>> f)
```
for which I'd expect two summaries:
```
Arg[this].SynthField[T] -> Arg[0].Param[0]
Arg[0].ReturnValue.Element -> ReturnValue.SynthField[T]
```
It's crucial here that the two type variables are different. Had they both been `T` then we probably also ought to get
```
Arg[0].ReturnValue.Element -> Arg[this].SynthField[T]
```

Reply:
I intentionally only made rudimentary support for higher order methods as I would like to do that as a follow up.